tests/Aspire.Hosting.Milvus.Tests/MilvusFunctionalTests.cs

```diff
@@ -0,0 +1,368 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Milvus.Client;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Aspire.Hosting.Milvus.Tests;
+
+public class MilvusFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    //Right now can not set user and password for super user of Milvus at startup. default user and password is root:Milvus.
+    //https://github.com/milvus-io/milvus/issues/33058
+    private const string MilvusToken = "root:Milvus";
+
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyMilvusResource()
+    {
+        var builder = CreateDistributedApplicationBuilder();
+
+        builder.Configuration["Parameters:apikey"] = MilvusToken;
+        var apiKey = builder.AddParameter("apikey");
+        var milvus = builder.AddMilvus("milvus", apiKey: apiKey);
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{milvus.Resource.Name}"] = await milvus.Resource.ConnectionStringExpression.GetValueAsync(CancellationToken.None)
+        });
+
+        hb.AddMilvusClient(milvus.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        var milvusClient = host.Services.GetRequiredService<MilvusClient>();
+
+        string collectionName = "book";
+        var collection = await milvusClient.CreateCollectionAsync(
```

Comment:
I think we will need some resilience added to this test as it is currently failing. Looks like it tries hitting the database before it is ready.